text/0031-configuration.md

```diff
@@ -0,0 +1,280 @@
++ Feature name: `configuration`
++ Start date: 2020-05-06
++ RFC PR: [iotaledger/bee-rfcs#31](https://github.com/iotaledger/bee-rfcs/pull/31)
++ Bee issue: [iotaledger/bee#71](https://github.com/iotaledger/bee/issues/71)
+
+# Summary
+
+This RFC proposes a configuration pattern for Bee binary and library crates.
+
+# Motivation
+
+This RFC contains a set of recommendations regarding configuration management in order to ensure consistency across the
+different Bee crates.
+
+# Detailed design
+
+## Libraries
+
+- [serde](https://github.com/serde-rs/serde): the go-to framework for **ser**ializing and **de**serializing Rust data
+  structures efficiently and generically;
+- [toml-rs](https://github.com/alexcrichton/toml-rs): a TOML encoding/decoding library for Rust with
+  serialization/deserialization on top of `serde`;
+
+## Recommendations
+
+With the following recommendations, one can:
+- read a configuration builder from a file;
```

Comment:
read a configuration builder from file? Isn't it either "read a configuration from file", or "create a configuration builder from file"?

Reply:
I prefer read because that's what's really happening. You read the initial state of the builder because there is a deserialization. You never "create" it by calling `new`/`default`.

Reply:
okay, just making sure, you can back it up. If so say it makes more sense, then I am fine with it.